File: c/242-valid-anagram.c
bool isAnagram(char* s, char* t) {
    if(strlen(s) != strlen(t)) return false;

    int count[26] = {0};

    for(int i=0; s[i] != '\0'; i++){
        count[s[i] - 97]++;
        count[t[i] - 97]--;
    }

    for(int i=0; i < 26; i++){
        if(count[i] != 0) return false;
    }

    return true;
}

File: js/242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {

    if(s.length !== t.length) return false;

    const freq = Array(26).fill(0);

    for(let i=0; i < s.length; i++){
        freq[s.charCodeAt(i)-97] += 1;
        freq[t.charCodeAt(i)-97] -= 1;
    }

    return !freq.some(item => item !== 0);
};
